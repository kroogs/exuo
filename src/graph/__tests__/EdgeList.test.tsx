/*
 * Copyright Â© 2020 Ty Dira <ty@dira.dev>

 * This file is part of Exuo.

 * Exuo is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.

 * Exuo is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.

 * You should have received a copy of the GNU Affero General Public License
 * along with Exuo.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react'
/* import { mount } from 'enzyme' */

/* import { EdgeList } from '../views/EdgeList' */
/* import { Node } from '../models/Node' */
/* import { graphFactory } from '../models/factories' */

/* let nextId = 0 */
/* const getId = (): string => String(++nextId) */

/* const Graph = graphFactory({ Node }, { getId }) */

test.skip('Renders a list of associated nodes', () => {
  /* const graph = Graph.create() */
  /* const first = graph.createNode() */
  /* const last = graph.createNode() */
  /* let prev = null */
  /* for (let i = 0; i < 5; i++) { */
  /*   const node = graph.createNode() */
  /*   node.addEdge('last', last) */
  /*   node.addEdge('first', first) */
  /*   first.addEdge('child', node) */
  /*   if (prev) { */
  /*     prev.addEdge('next', node) */
  /*     node.addEdge('prev', prev) */
  /*   } */
  /*   prev = node */
  /* } */
  /* expect(mount(<EdgeList edgeTag="child" node={first} />)).toMatchSnapshot() */
})
