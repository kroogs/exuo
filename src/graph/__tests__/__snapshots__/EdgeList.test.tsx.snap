// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a list of associated nodes 1`] = `
<EdgeList
  node={
    Object {
      "edgeMap": Object {
        "child": Array [
          "3",
          "4",
          "5",
          "6",
          "7",
        ],
      },
      "id": "1",
      "label": "",
    }
  }
>
  <WithStyles(ForwardRef(List))
    aria-label="edge list"
    className="makeStyles-list-1"
  >
    <ForwardRef(List)
      aria-label="edge list"
      className="makeStyles-list-1"
      classes={
        Object {
          "dense": "MuiList-dense",
          "padding": "MuiList-padding",
          "root": "MuiList-root",
          "subheader": "MuiList-subheader",
        }
      }
    >
      <ul
        aria-label="edge list"
        className="MuiList-root makeStyles-list-1 MuiList-padding"
      />
    </ForwardRef(List)>
  </WithStyles(ForwardRef(List))>
</EdgeList>
`;
